cpu/esp8266/Makefile.features

```diff
@@ -0,0 +1,10 @@
+FEATURES_PROVIDED += periph_adc
```

Comment:
are these *always* configured and available?

RIOT convention is to keep *configureable* features in periph_conf.h. E.g., most MCU's provide a handful of SPI blocks, but most boards configure only some or none (when, for example, the corresponding pins are not connected at all). That's why ```periph_spi``` is provided in ````$BOARD/Makefile.features```.

So if it is possible (or even the case) that some of these features are actually not configured in some boards, these provided features have to be moved to board level.

Reply:
Yes, the only one ADC channel is always available. It is a dedicated MCU pin and is broken out on all boards I know, except the ESP-01.

As explained for SPI, all ESP8266 boards simply break out all the GPIOs of the MCU. The software decides, how the GPIOs are used, not the board. All features are always available.

Reply:
I think this is still an issue. In RIOT, whether a board "supports" a feature is a function of hardware capabilities and software configuration. That is why "boards" define features.

The question would be if e.g., "GPIO2" is *always* configured as I2C or if, by changing "perihp_conf.h", it can be configured to e.g., SPI and then no I2C is available. That's how it is for most MCUs, and in that case the feature definition must move to the board.

Reply:
In the case of ADC pin of the ESPP8266 it seems to be different. It is nothing the software can configure or affect. The ADC pin is no GPIO pin. It is a dedicated pin which is always active and cannot be used for any other purpose, and it is always broken out. IMHO, it is completly provided by the MCU.

Anyway, I have already restructered the Makefiles further (just not committed) and
```
FEATURES_PROVIDED += periph_adc
```
is already part of ```boards/common/esp8266/Makefile.feature.